packages/orbit-components/src/CountryFlag/index.tsx

```diff
@@ -1,77 +1,52 @@
-import { convertHexToRgba } from "@kiwicom/orbit-design-tokens";
+import cx from "clsx";
 
-import type { Theme } from "../defaultTheme";
-import defaultTheme from "../defaultTheme";
-import { baseURL, CODES, SIZES, TOKENS } from "./consts";
-import type { Props, Size } from "./types";
+import { baseURL, CODES, SIZE_WIDTHS, SIZES } from "./consts";
+import type { Props } from "./types";
 
-const getSizeToken =
-  (name: string) =>
-  ({ theme, size }: { theme: Theme; size: Size }) => {
-    const tokens = {
-      [TOKENS.WIDTH]: {
-        [SIZES.SMALL]: "16px",
-        [SIZES.MEDIUM]: theme.orbit.widthCountryFlag,
-      },
-      [TOKENS.HEIGHT]: {
-        [SIZES.SMALL]: "9px",
-        [SIZES.MEDIUM]: "13px",
-      },
-    };
-    return tokens[name][size];
-  };
-
-const StyledCountryFlag = styled.div<{ size: Size }>`
-  position: relative;
-  height: ${getSizeToken(TOKENS.HEIGHT)};
-  width: ${getSizeToken(TOKENS.WIDTH)};
-  background-color: ${({ theme }) => theme.orbit.backgroundCountryFlag};
-  border-radius: ${({ theme }) => theme.orbit.borderRadiusSmall};
-  overflow: hidden;
-  flex-shrink: 0;
-`;
-
-StyledCountryFlag.defaultProps = {
-  theme: defaultTheme,
-};
-
-export const StyledImage: any = styled.img.attrs<{ size: Size; code: string }>(
-  ({ theme, size, code }: { theme: Theme; size: Size; code: string }) => {
-    const width = parseInt(getSizeToken(TOKENS.WIDTH)({ theme, size }), 10);
-    return {
-      src: `${baseURL}/flags/${width}x0/flag-${code.toLowerCase()}.jpg`,
-      srcSet: `${baseURL}/flags/${width * 2}x0/flag-${code.toLowerCase()}.jpg 2x`,
-    };
-  },
-)`
-  display: block;
-  height: 100%;
-  width: 100%;
-  flex-shrink: 0;
-`;
-
-StyledImage.defaultProps = {
-  theme: defaultTheme,
+const Wrapper = ({ children, size }) => {
+  return (
+    <div
+      className={cx("relative bg-country-flag-background rounded-small overflow-hidden shrink-0", {
+        "w-country-flag-small h-country-flag-small": size === SIZES.SMALL,
+        "w-country-flag-medium h-country-flag-medium": size === SIZES.MEDIUM,
+      })}
+    >
+      {children}
+    </div>
+  );
 };
 
-const StyledShadow = styled.div`
-  position: absolute;
-  display: block;
-  height: 100%;
-  width: 100%;
-  top: 0;
-  right: 0;
-  bottom: 0;
-  left: 0;
-  box-shadow: inset 0 0 0 1px ${({ theme }) => convertHexToRgba(theme.orbit.paletteInkDark, 10)};
-  border-radius: ${({ theme }) => theme.orbit.borderRadiusSmall};
-`;
+const Image = ({
+  code,
+  name,
+  id,
+  dataTest,
+  src,
+  srcSet,
+}: {
+  code: Props["code"];
+  name: Props["name"];
+  id: Props["id"];
+  dataTest: Props["dataTest"];
+  src: string;
+  srcSet: string;
+}) => (
+  <img
+    className="block h-full w-full shrink-0"
+    key={code}
+    alt={name}
+    title={name}
+    id={id}
+    data-test={dataTest}
+    src={src}
+    srcSet={srcSet}
+  />
+);
 
-StyledShadow.defaultProps = {
-  theme: defaultTheme,
-};
+const Shadow = () => (
```

Comment:
I think no need to have a component for that, you can directly use it as div with className inside CountryFlag :)

Reply:
Yeah I gave it a thought about that while implementing for the first time. I decided to keep the abstraction to improve readability. Having a div with a bunch of classes to apply a shadow makes it a bit hard to read / understand what that is. For the image I can agree and I am ok with removing the abstraction. But with the shadow… idk. I can accept either decision, but please consider the argument of readability and let me know what you think 🙂

Reply:
I do not think it's making it worse to read :) 

It would, in the case of the bigger component, but for that case, it looks ok to me to write directly there :)

Reply:
I will leave decision up to you, I do not want to insist on something like that :)

Reply:
Yes, going through it, I think we can remove all abstraction. I will move the `shadow-country-flag` class to the start of the component, so it is easier to spot and understand that that div refers to a shadow 🙂